risc0/zkvm/src/host/api/tests.rs

```diff
@@ -239,6 +240,109 @@ fn lift_join_identity() {
     rollup_receipt.verify(MULTI_TEST_ID).unwrap();
 }
 
+#[cfg(feature = "docker")]
+#[test]
+fn stark2snark() {
+    const SEAL_FILE: &str = "seal.bin";
+
+    let cycles = 0u32;
+    let env = ExecutorEnv::builder()
+        .write(&MultiTestSpec::BusyLoop { cycles })
+        .unwrap()
+        .build()
+        .unwrap();
+    let binary = Asset::Inline(MULTI_TEST_ELF.into());
+
+    let mut client = TestClient::new();
+
+    let session = client.execute(env, binary);
+    assert_eq!(session.segments.len(), client.segments.len());
+
+    let opts = ProverOpts::default();
+
+    let receipt = client.prove_segment(opts.clone(), client.segments[0].clone());
+    let mut rollup = client.lift(opts.clone(), receipt.try_into().unwrap());
+
+    for segment in &client.segments[1..] {
+        let receipt = client.prove_segment(opts.clone(), segment.clone());
+        let rec_receipt = client.lift(opts.clone(), receipt.try_into().unwrap());
+
+        rollup = client.join(
+            opts.clone(),
+            rollup.try_into().unwrap(),
+            rec_receipt.try_into().unwrap(),
+        );
+        rollup
+            .verify_integrity_with_context(&VerifierContext::default())
+            .unwrap();
+    }
+    let receipt_ident = client.identity_p254(opts, rollup.clone().try_into().unwrap());
+
+    let work_dir = tempdir().expect("Failed to create tmpdir");
+    let seal_path = work_dir.path().join(SEAL_FILE);
+    std::fs::write(&seal_path, &receipt_ident.get_seal_bytes())
+        .expect("Failed to write seal-to-json stdout to disk");
+
+    let journal = session.journal.bytes;
+
+    let rollup_receipt = Receipt::new(InnerReceipt::Succinct(rollup), journal.clone());
+    rollup_receipt.verify(MULTI_TEST_ID).unwrap();
+
+    let output = Command::new("docker")
+        .arg("run")
+        .arg("--rm")
+        .arg("-v")
+        .arg(&format!(
+            "{:}:/app/seal.bin:ro",
+            seal_path.to_string_lossy()
+        ))
+        .arg("angelocapossele/risc0-groth16-prover:v0.0.1")
```

Comment:
The repo for this docker image should change if we decide to take the docker route. This is only for testing purposes